src/mds/Server.cc

```diff
@@ -5696,10 +5696,19 @@ bool Server::xlock_policylock(const MDRequestRef& mdr, CInode *in, bool want_lay
 CInode* Server::try_get_auth_inode(const MDRequestRef& mdr, inodeno_t ino)
 {
   CInode *in = mdcache->get_inode(ino);
-  if (!in || in->state_test(CInode::STATE_PURGING)) {
+  if (in && in->state_test(CInode::STATE_PURGING)) {
     respond_to_request(mdr, -CEPHFS_ESTALE);
     return nullptr;
   }
+  /*
```

Comment:
nit: possibly move this code block above the `if (in && .._)`. That would avoid checking `in` again.

Reply:
Sure, will fix it.